jena-core/src/main/java/org/apache/jena/ext/xerces/impl/validation/ValidationState.java

```diff
@@ -86,11 +85,25 @@ public void setSymbolTable(SymbolTable sTable) {
         fSymbolTable = sTable;
     }
 
+    /**
+     * Ensures that fIdTable and fIdRefTable are initialized.
+     */
+    private void ensureIdTablesAreInitialized() {
+        if (fIdTable == null) {
+            // REVISIT: Should replace with a lighter structure.
+            fIdTable = new HashMap();
+        }
+        if (fIdRefTable == null) {
+            fIdRefTable = new HashMap();
+        }
+    }
+
     /**
      * return null if all IDREF values have a corresponding ID value;
      * otherwise return the first IDREF value without a matching ID value.
      */
     public String checkIDRefID () {
+        ensureIdTablesAreInitialized();
```

Comment:
Why init here? Instead:
`if(fIdRefTable == null) return null;`

and later in the while loop:

 if (fIdTable == null || !fIdTable.containsKey(key)) {